### PR TITLE
Add waiting state to showcase form

### PR DESCRIPTION
**Pull request by dipamsen**
See #1550 

- Adds "waiting" state to showcase form to prevent duplicate submission from double clicks
- Style `disabled` button properly 
 
cc @fturmel

### CODE REVIEW

File: src/components/Button.module.css
Comment (fturmel) [inline src/components/Button.module.css:1]: Can we use the pseudo class `:disabled` instead of the attribute `disabled`? We might not need all the color variants specificity here either.

```css
.rainbow:hover:not(:disabled) {
  background-position: left bottom;
  transition: background-position 4s ease-out;
}

button:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}
```

And well done on the negation here, I was wondering why you didn't use `:enabled` initially but this button component is sneaky and can end up being a `<button>` or a `<Link>` / `<a>` tag based on the props passed to it.

File: src/components/PassengerShowcaseForm.module.css [hunk 1]
```diff
@@ -64,6 +64,6 @@
   }
 }
 
-.submitBtn:hover {
+.submitBtn:hover:not([disabled]) {
```
Comment (fturmel) [inline src/components/PassengerShowcaseForm.module.css:67]: Here we could actually use `:enabled` since it is always a button

**Comment by shiffman** [comment]
Looks great, thank you @dipamsen! Tagging in @fturmel in case you have a chance to review!

File: src/components/PassengerShowcaseForm.js [hunk 2]
```diff
@@ -429,7 +435,7 @@ const PassengerShowcaseForm = () => {
           className={css.submitBtn}
           onClick={onSubmit}
           variant="purple"
-          disabled={submitted}
+          disabled={submitted !== 'not-submitted'}
```
Comment (fturmel) [inline src/components/PassengerShowcaseForm.js:438]: Nit-pick but I find it hard to parse cognitively with the double (triple?) negative.

Maybe renaming the variable to `submissionState` and finding an alternate name for `not-submitted` state that doesn't contain a negative would help? What about `undefined`, would that read better?

```ts
submissionState === 'submitted'

disabled = !!submissionState
```

**Comment by fturmel** [comment]
Looks great, I agree!

Sorry for not shipping a fix myself after self-assigning the issue. I've been taking notes on additional improvements to the submission process and did not get a chance to code or share anything yet. More soon-ish!

File: src/components/Button.module.css
Comment (dipamsen) [reply src/components/Button.module.css:1]: whoops, didn't realise i was inconsistently using `:disabled` and `[disabled]` in different places

Comment (fturmel) [reply src/components/Button.module.css:1]: Thought about this a bit more, and maybe the `:not(:disabled)` bit is too "clever" without a comment? Maybe something like this would be clearer/more self-documenting in the long run.

```css
a.rainbow, button.rainbow:enabled {
  &:hover {
    background-position: left bottom;
    transition: background-position 4s ease-out;
  }
}
```

**Comment by shiffman** [comment]
Thank you @dipamsen, excited to merge this!

**Comment by dipamsen** [comment]
Oh whoops looks like the review comments were pending to be addressed...

**Comment by shiffman** [comment]
Oh, sorry about that! Would you like me to revert or shall we just start a new branch? Apologies for this mistake!

**Comment by fturmel** [comment]
No worries, and no need to revert since the comments were more about minor code quality improvements. I can address them in an upcoming PR.
